circleguard/comparer.py

```diff
@@ -144,7 +146,35 @@ def _compare_two_replays(replay1, replay2):
         if (Mod.HR in replay1.mods) ^ (Mod.HR in replay2.mods):
             xy1[:, 1] = 384 - xy1[:, 1]
 
-        return Comparer._compute_data_similarity(xy1, xy2)
+        (play_sequence_1, play_sequence_2) = Comparer.convert_to_row_major_form(xy1, xy2)
+
+        NUM_CHUNKS = 20
+
+        # Sectioned into 20 chunks, used to ignore outliers (eg. long replay with breaks is copied, and the cheater cursordances during the break)
+        horizontal_length = play_sequence_1.shape[1] - play_sequence_1.shape[1] % NUM_CHUNKS
+        play_sequence_1_sections = np.hsplit(play_sequence_1[:,:horizontal_length], NUM_CHUNKS)
+        play_sequence_2_sections = np.hsplit(play_sequence_2[:,:horizontal_length], NUM_CHUNKS)
+        correlations = []
+        for (play_sequence_1, play_sequence_2) in zip(play_sequence_1_sections, play_sequence_2_sections):
+            cross_correlation_matrix = Comparer.find_cross_correlation(play_sequence_1, play_sequence_2)
+            #Pick the lag with the maximum correlation, this likely in most cases is 0 lag
+            max_correlation = max(cross_correlation_matrix.reshape(cross_correlation_matrix.size))
```

Comment:
is this reshape necessary if you use numpy's max?

Reply:
max might be good to use but unless there's a good reason against it, consider using a mean, as max has an upward bias and might give some unfortunate outliers

Reply:
Hmm, maybe my understanding of the math is wrong, but this what I'm basing it off.

https://www.statisticshowto.com/cross-correlation/

The max of the cross correlation is where the two signals correlate the best. The mean of the cross correlation matrix will almost always be a very low number afaik.

Imagine if you get the correlations of sin(t) and sin(t - n), so you end up with two signals that are the same but out of phase. You'll get a normalized correlation matrix where the mean will be 0 despite the signals being highly correlated. Picking the max from the cross correlation matrix will give you a correlation of 1, telling you that the signals are the same, but shifted in the time dimension.

I could be wrong though, honestly, haven't done any math in years.

I think the outlier problem is actually solved by picking the median out of all the max correlations from every single slice?

Reply:
you are correct and you should use max (it doesn't introduce bias because it's the only useful value to inspect), I had mistakenly imagined the correlation as being instantaneous but it is not and that's the only reason you need to even divide the replay. 
The way you do correlation should also filter shifts in the xy-plane correct?

Reply:
Yes, it should catch translations in the xy plane.